app/src/main/java/com/yalta/repositories/LocationRepo.kt

```diff
@@ -0,0 +1,36 @@
+package com.yalta.repositories
+
+import android.location.Location
+import com.yalta.services.SessionService
+import common.LocationUpdate
+import okhttp3.*
+
+sealed class LocationResponse
+class AddedLocation(val location: common.Location) : LocationResponse()
+class FailedLocation : LocationResponse()
```

Comment:
Neat! I'd work a little on naming cause it's not really coherent, but the idea is really great! 
Maybe we can expand this on everything! I mean: create superclass `Response` that is inherited by `Error` and `SuccessfulResponse`. And every Entity that can be sent as response (or specific responses) will just extend this one! 
This will allow generalizing all http-calling method return values to Response and simplify all checks! 
We can base deserialization on http response codes and can even not include base types into json!

Reply:
Created issue